File: 633.平方数之和.go
/*
 * @lc app=leetcode.cn id=633 lang=golang
 *
 * [633] 平方数之和
 */
package leetcode

import "math"

// @lc code=start
func judgeSquareSum(c int) bool {
	i, j := 0, int(math.Sqrt(float64(c)))
	for i <= j {
		s := i*i + j*j
		if s == c {
			return true
		}
		if s < c {
			i++
		} else {
			j--
		}
	}
	return false
}

// @lc code=end

File: 633.平方数之和.cpp
/*
 * @lc app=leetcode.cn id=633 lang=cpp
 *
 * [633] 平方数之和
 */

#include "corecrt_math.h"
#include <vcruntime.h>

// @lc code=start
class Solution {
public:
  bool judgeSquareSum(int c) {
    size_t i = 0;
    size_t j = sqrt(c);
    while (i <= j) {
      size_t s = i * i + j * j;

      if (s == c) {
        return true;
      }

      if (s < c) {
        i++;
      } else {
        j--;
      }
    }
    return false;
  }
};
// @lc code=end
